aretomo/protocols/protocol_aretomo.py

```diff
@@ -457,19 +463,33 @@ def runAreTomoStep(self, tsId: str, tsFn: str):
             self.error('Aretomo execution failed for tsId %s -> %s' % (tsId, e))
 
     def createOutputStep(self, tsId: str, tsFn: str):
-        if tsId in self._failedTsList:
-            self.createOutputFailedTs(tsId)
-        else:
-            self.createOutputTs(tsId, tsFn)
+        with self._lock:
+            if tsId in self._failedTsList:
+                self.createOutputFailedTs(tsId)
+            else:
+                self.createOutputTs(tsId, tsFn)
+            for outputName in self._possibleOutputs.keys():
```

Comment:
why closing output here? There's a separate step closeOutputSetStep that does this

Reply:
Hi, @azazellochg,
That is the solution we've found for some concurrency problems that were still happening. We'll may change this in future, but for now it seems to be the best solution to prevent those problems.

Reply:
Ok :) If it works better then great! Then we can remove  closeOutputSetStep?

Reply:
No, it has to be there. It's like we have to close each generated output each time the protocol opens and write them while working. Maybe @oierlauzi or @pconesa can provide a better explanation, as they were the ones that noticed that out and proposed this solution.